providers/implementations/kem/template_kem.c

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright 2024 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <string.h>
+#include <openssl/crypto.h>
+#include <openssl/evp.h>
+#include <openssl/core_dispatch.h>
+#include <openssl/core_names.h>
+#include <openssl/params.h>
+#include <openssl/err.h>
+#include <openssl/proverr.h>
+#include "prov/provider_ctx.h"
+#include "prov/implementations.h"
+#include "prov/securitycheck.h"
+#include "prov/providercommon.h"
+
+extern const OSSL_DISPATCH ossl_template_asym_kem_functions[];
+
+#ifdef NDEBUG
```

Comment:
This will probably have to be `#if defined(NDEBUG) || defined(OPENSSL_NO_STDIO)`

Reply:
ok to resolve?